crowbar_framework/app/models/service_object.rb

```diff
@@ -1146,15 +1157,23 @@ def apply_role(role, inst, in_queue)
         end
       end # roles.each
 
-      @logger.debug "nodes_in_batch #{nodes_in_batch.inspect}"
-      run_order << nodes_in_batch unless nodes_in_batch.empty?
-      @logger.debug "run_order #{run_order.inspect}"
+      batches << nodes_in_batch unless nodes_in_batch.empty?
+    end
+    @logger.debug "batches: #{batches.inspect}"
+
+    # Cache attributes that are useful later on
+    pre_cached_nodes.each do |node_name, node|
```

Comment:
can we add the line `next if node.nil?` here as well, to skip nodes, that no longer exists, but still referenced in chef.